TileRarities.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+    The data for the different chances and multiplier values can be found on Data:Tile_Chances
+]]--
+local p = {}
+
+function p.main(frame)
+	return p._main(frame.args)
+end
+
+function p._main(args)
+	local json = mw.loadJsonData('Data:Tile_Chances')
+    local ranks = json.ranks
+	local dom = mw.html.create()
+	local tbl = dom:tag('table'):addClass('wikitable'):css('text-align', 'right')
+	local row = tbl:tag('tr')
+	row:tag('th'):attr('rowspan', 2):wikitext('Rarity'):css('vertical-align', 'bottom')
```

Comment:
i would probably pull this all out into a function. call `h.printHeader(dom)`, and do everything to do with printing the header there

note: im referring to this as Comment A in 2 other places where im suggesting changes related to this

Reply:
Not 100% sure what this would look like... Like I would assume moving all the stuff from `local dom` to `row:tag('th')...` into its own function and then have it return the row to use?

Reply:
No, I am only referring to creating the header row in its own function.

You would create the dom node in the parent function, and then you would send it to a helper function in which you create the header row and print all of the header cells. Then the header row is never used again for the entire rest of the code block, so you can leave the entire header row isolated in that header function and then it is a bit clearer what is going on with your entire logic

Reply:
Main issue I see here is the fact that not all tables have the exact same headers.
Some have header cells with `colspan="2"`, others with `colspan="3"`, etc. Also, the first header is not guaranteed to always only be `rowspan="2"` (See the sunday table on the tiles wiki page, where it is 3)
I obviously can make the function accept multiple args - dom, rowspan and colspan - but at least for rowspan would this feel like pointless extra fluff here.

Reply:
That is indeed what I meant. You are welcome to ignore my advice